tests/app/main/test_suppliers.py

```diff
@@ -2282,3 +2284,152 @@ def test_existing_org_size_sets_current_selection(self, data_api_client, existin
+
+            assert res.status_code == 302
+            assert res.location == "http://localhost/suppliers/details"
+            self.data_api_client.update_supplier.assert_called_once_with(
+                supplier_id=1234,
+                supplier={"vatNumber": "GB123456789"},
+                user="email@email.com"
+            )
+
+    @pytest.mark.parametrize('vat_number', ("", "Not a VAT number"))
+    def test_posting_not_vat_registered_correctly_updates_supplier(self, vat_number):
+        with self.app.test_client():
+            res = self.client.post(
+                "/suppliers/vat-number/edit",
+                data={
+                    "vat_registered": "No",
+                    "vat_number": vat_number,
+                }
+            )
+
+            assert res.status_code == 302
+            assert res.location == "http://localhost/suppliers/details"
+            self.data_api_client.update_supplier.assert_called_once_with(
+                supplier_id=1234,
+                supplier={"vatNumber": "Not VAT registered"},
+                user="email@email.com",
+            )
+
+    @pytest.mark.parametrize("vat_number, message, masthead", (
+        ("GB123456789", None, None),
+        ("gb123456789012", None, None),
+        ("gBGD123", None, None),
+        ("gbHA456", None, None),
+        ("123456789", None, None),
+        ("123456789012", None, None),
+        ("GD123", None, None),
+        ("Ha456", None, None),
+        ("NOTVALID", "You must provide a valid VAT number, they are usually either 9 or 12 digits.", "VAT number"),
+        ("", "You must provide a VAT number.", "VAT number")
+    ))
+    def test_validation_of_tax_number(self, vat_number, message, masthead):
+        with self.app.test_client():
+            res = self.client.post(
+                "/suppliers/vat-number/edit",
+                data={
+                    "vat_registered": "Yes",
+                    "vat_number": vat_number,
+                }
+            )
+            doc = html.fromstring(res.get_data(as_text=True))
+
+            if message:
+                assert res.status_code == 400
+                assert doc.xpath(
+                    "//div[@class='validation-masthead'][normalize-space(string())=$masthead]",
+                    masthead=f"There was a problem with your answer to: {masthead}"
+                )
+                assert doc.xpath(
+                    "//span[@class='validation-message'][normalize-space(string())=$message]",
+                    message=message
+                )
+                assert not self.data_api_client.update_supplier.called
+            else:
+                assert res.status_code == 302
+                self.data_api_client.update_supplier.assert_called_once_with(
+                    supplier_id=1234,
+                    supplier={"vatNumber": vat_number},
+                    user="email@email.com",
+                )
+
+    def test_validation_of_vat_registered(self):
```

Comment:
the name of this test could be more specific, like `test_validation_against_empty_string_for_vat_registered_suppliers`